compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3766,69 +3806,100 @@ abstract class GenJSCode extends plugins.PluginComponent
           })(jstpe.AnyType)
         }
+
+        /* For primitive types, we need to handle two cases. The method could
+         * either be defined in the boxed class of the primitive type, or it
+         * could be defined in the primitive class itself. In the former case,
+         * we generate a direct call to the actual (hijacked) method. In the
+         * latter case, we directly generate the primitive operation.
+         */
+
+        def addCallForPrimitive(primitiveClass: Symbol): Boolean = {
+          val boxedClass = definitions.boxedClass(primitiveClass)
+          val boxedTpe = boxedClass.tpe
+          val methodInBoxedClass = matchingSymIn(boxedClass)
+          if (methodInBoxedClass != NoSymbol && methodInBoxedClass.isPublic) {
+            callStatement = js.If(genIsInstanceOf(callTrg, boxedTpe), {
+              boxIfNeeded(
+                  genApplyMethod(
+                      genAsInstanceOf(callTrg, boxedTpe),
+                      methodInBoxedClass,
+                      arguments),
+                  methodInBoxedClass.tpe.resultType)
+            }, { // else
+              callStatement
+            })(jstpe.AnyType)
+            true
+          } else {
+            val methodInPrimClass = matchingSymIn(primitiveClass)
+            if (methodInPrimClass != NoSymbol && methodInPrimClass.isPublic) {
+              def isIntOrLongKind(kind: TypeKind) = kind match {
+                case CharKind     => false
+                case _:INT | LONG => true
+                case _            => false
+              }
+              val ignoreBecauseItMustBeAnInt = {
+                primitiveClass == DoubleClass &&
+                toTypeKind(methodInPrimClass.tpe.resultType) == DoubleKind &&
+                isIntOrLongKind(toTypeKind(sym.tpe.resultType))
+              }
+              if (ignoreBecauseItMustBeAnInt) {
+                // Fall through to the Int case that will come next
+                false
+              } else {
+                callStatement = js.If(genIsInstanceOf(callTrg, boxedTpe), {
+                  val castCallTrg =
+                    makePrimitiveUnbox(callTrg, primitiveClass.tpe)
+                  val call = genPrimitiveOpForReflectiveCall(methodInPrimClass,
+                      castCallTrg, arguments)
+                  boxIfNeeded(call, methodInPrimClass.tpe.resultType)
+                }, { // else
+                  callStatement
+                })(jstpe.AnyType)
+                true
+              }
+            } else {
+              false
+            }
+          }
+        }
+
+        addCallForPrimitive(BooleanClass)
+        addCallForPrimitive(LongClass)
+        addCallForPrimitive(CharClass)
+
+        /* For primitive numeric types that box as JS numbers, find the first
+         * one that matches. It will be able to handle the subsequent cases.
+         */
+        Seq(DoubleClass, IntClass, FloatClass, ShortClass, ByteClass).find(
```

Comment:
Yes, even a strict `Float` is first matched as a `Double`, so just like for `Int`s, `Short`s and `Byte`s, most operations are handled by the `Double` case. `Int` has a few more cases to itself, typically bitwise operations. The only thing that's left for `Float`, `Short` and `Byte` is their specific `compareTo` method.